src/bin/dual-iir.rs

```diff
@@ -425,26 +457,118 @@ mod app {
+            ) = (pounder, settings.pounder, c.local.dds_clock_state)
+            {
+                if *clocking != pounder_settings.dds_clock {
+                    *clocking = pounder_settings.dds_clock;
+                    pounder_devices
+                        .set_ext_clk(pounder_settings.dds_clock.external_clock)
+                        .unwrap();
+
+                    ddses
+                        .builder()
+                        .set_system_clock(
+                            pounder_settings.dds_clock.reference_clock,
+                            pounder_settings.dds_clock.multiplier,
+                        )
+                        .unwrap()
+                        .write();
+                }
+
+                let system_clock_frequency =
+                    pounder_settings.dds_clock.reference_clock
+                        * pounder_settings.dds_clock.multiplier as f32;
+
+                for (channel_config, pounder_channel) in pounder_settings
+                    .in_ch
+                    .iter()
+                    .chain(pounder_settings.out_ch.iter())
+                    .zip([
+                        PounderChannel::In0,
+                        PounderChannel::In1,
+                        PounderChannel::Out0,
+                        PounderChannel::Out1,
+                    ])
+                {
+                    let dds_profile = channel_config
+                        .dds
+                        .try_into_dds_profile(system_clock_frequency)
+                        .unwrap();
+                    ddses
+                        .builder()
+                        .update_channels_with_profile(
+                            pounder_channel.into(),
+                            dds_profile,
+                        )
+                        .write();
+
+                    if pounder_devices.get_attenuation(pounder_channel).unwrap()
+                        != channel_config.attenuation
+                    {
+                        pounder_devices
+                            .set_attenuation(
+                                pounder_channel,
+                                channel_config.attenuation,
+                            )
+                            .unwrap();
+                    }
+                }
+            }
+        });
+
         let target = settings.stream_target.into();
         c.shared.network.lock(|net| net.direct_stream(target));
     }
 
-    #[task(priority = 1, shared=[network, settings, telemetry], local=[cpu_temp_sensor])]
+    #[task(priority = 1, shared=[network, settings, telemetry, pounder], local=[cpu_temp_sensor])]
     fn telemetry(mut c: telemetry::Context) {
-        let telemetry: TelemetryBuffer =
+        let mut telemetry: TelemetryBuffer =
             c.shared.telemetry.lock(|telemetry| *telemetry);
 
+        telemetry.cpu_temp = c.local.cpu_temp_sensor.get_temperature().unwrap();
+        telemetry.pounder = c.shared.pounder.lock(|pounder| {
```

Comment:
This is convoluted. By default `telemetry.pounder` is `None`. No need to reset it all the time.
And can't this code go into `PounderTelemetry` or `PounderDevices` so that you don't have to copy it everywhere?